rllib/algorithms/dqn/dqn_rainbow_rl_module.py

```diff
@@ -0,0 +1,185 @@
+            self.v_min: float = self.config.model_config_dict.get("v_min")
+            self.v_max: float = self.config.model_config_dict.get("v_max")
+
+        # Build the encoder for the advantage and value streams. Note,
+        # the same encoder is used.
+        # Note further, by using the base encoder the correct encoder
+        # is chosen for the observation space used.
+        self.encoder = catalog.build_encoder(framework=self.framework)
+        # Build the same encoder for the target network(s).
+        self.target_encoder = catalog.build_encoder(framework=self.framework)
+
+        # Build heads.
+        self.af = catalog.build_af_head(framework=self.framework)
+        if self.is_dueling:
+            self.vf = catalog.build_vf_head(framework=self.framework)
+        # Implement the same heads for the target network(s).
+        self.af_target = catalog.build_af_head(framework=self.framework)
+        if self.is_dueling:
+            self.vf_target = catalog.build_vf_head(framework=self.framework)
+
+        # We do not want to train the target networks.
+        self.target_encoder.trainable = False
+        self.af_target.trainable = False
+        if self.is_dueling:
+            self.vf_target.trainable = False
+
+        # Define the action distribution for sampling the eexploit action
+        # during exploration.
+        self.action_dist_cls = catalog.get_action_dist_cls(framework=self.framework)
+
+    @override(RLModule)
+    def get_exploration_action_dist_cls(self) -> Type[Distribution]:
+        """Returns the action distribution class for exploration.
+
+        Note, this class is used to sample the exploit action during
+        exploration.
+        """
+        return self.action_dist_cls
+
+    # TODO (simon): DQN Rainbow does not support RNNs, yet.
+    @override(RLModule)
+    def get_initial_state(self) -> Any:
+        return {}
+
+    @override(RLModule)
+    def input_specs_exploration(self) -> SpecType:
+        return [SampleBatch.OBS]
+
+    @override(RLModule)
+    def input_specs_inference(self) -> SpecType:
+        return [SampleBatch.OBS]
+
+    @override(RLModule)
+    def input_specs_train(self) -> SpecType:
+        return [
+            SampleBatch.OBS,
+            SampleBatch.ACTIONS,
+            SampleBatch.NEXT_OBS,
+        ]
+
+    @override(RLModule)
+    def output_specs_exploration(self) -> SpecType:
+        return
+
+    @abstractmethod
+    @OverrideToImplementCustomLogic
+    def _qf(self, batch: Dict[str, TensorType]) -> Dict[str, TensorType]:
+        """Computes Q-values.
+
+        Note, these can be accompanied with logits and pobabilities
+        in case of distributional Q-learning, i.e. `self.num_atoms > 1`.
+
+        Args:
+            batch: The batch recevied in the forward pass.
+
+        Results:
+            A dictionary containing the Q-value predictions and in case of
+            distributional Q-leanring the support atoms ("atoms"), the
```

Comment:
```suggestion
            distributional Q-learning the support atoms ("atoms"), the
```

Reply:
Can you add, what the exact str key should be for non-distributional DQN? And then say that for distribution DQN, that key is not needed, but these other three are required?